### PR TITLE
Enable nullability in remaining ControlDesigner members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in remaining ControlDesigner members.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11513)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 1]
```diff
@@ -2312,7 +2312,7 @@ private void PaintException(PaintEventArgs e, Exception ex)
         stringFormat.Dispose();
     }
 
-    private IOverlayService OverlayService => _overlayService ??= GetService<IOverlayService>();
+    private IOverlayService? OverlayService => _overlayService ??= GetService<IOverlayService>();
```
Comment (ricardobossan) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:2315]: Use TryGet pattern, to ensure non-null return values, like this:

```cs
   private IOverlayService? OverlayService
   {
       get
       {
           if (_overlayService is null)
           {
               if (TryGetService<IOverlayService>(out var service))
               {
                   _overlayService = service;
               }
           }

           return _overlayService;
       }
   }
```

Comment (gpetrou) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:2315]: Not sure I understand the suggestion. This is still nullable.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 2]
```diff
@@ -2490,8 +2490,8 @@ internal void SetUnhandledException(Control owner, Exception exception)
         _thrownException = exception;
         owner ??= Control;
 
-        string[] exceptionLines = exception.StackTrace.Split('\r', '\n');
-        string typeName = owner.GetType().FullName;
+        string[] exceptionLines = exception.StackTrace!.Split('\r', '\n');
+        string typeName = owner.GetType().FullName!;
```
[the remark this answers is not in the document]
Comment (gpetrou) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:2494]: I doubt that these will ever be null, but I added a couple of null checks anyway.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs (hunk 1 above)
Comment (ricardobossan) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:2315]: We recommend the TryGet pattern, like TryGetService, to ensure non-null return values and maintain consistency. While the value can still be nullable, this approach standardizes our code. It’s encouraged for consistency but not mandatory.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.ChildSubClass.cs [hunk 3]
```diff
@@ -58,7 +58,7 @@ protected override void WndProc(ref Message m)
             }
             catch (Exception ex)
             {
-                _designer.SetUnhandledException(Control.FromChildHandle(m.HWnd), ex);
+                _designer.SetUnhandledException(Control.FromChildHandle(m.HWnd)!, ex);
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.ChildSubClass.cs:61]: Looking at `SetUnhandledException` seems like it is ok for first param to be null. We should adjust the signature in that method and avoid the `!` here

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs (hunk 1 above)
Comment (lonitra) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:2315]: We typically use `TryGetService<T>` if we are going to do different logic based off of whether we can get a non-null service or not. In this case, what we return is still nullable so leaving this as `GetService<T>` is acceptable.

**Review by lonitra — APPROVED** [review]
Thank you!

**Pull request merged**